h2o-core/src/main/java/hex/Model.java

```
@@ -1277,8 +1281,12 @@ else if (isWeights && computeMetrics) {
       }
       vvecs[i] = vec;
     }
+
+    if( good == convNaN )
+      throw new IllegalArgumentException("Test/Validation dataset has no columns in common with the training set");
+
```

fail early, otherwise another weaker condition may return earlier